variant.go

```diff
@@ -17,8 +17,8 @@ type VariantService interface {
 	List(int64, interface{}) ([]Variant, error)
 	Count(int64, interface{}) (int, error)
 	Get(int64, interface{}) (*Variant, error)
-	Create(int64, Variant) (*Variant, error)
-	Update(Variant) (*Variant, error)
+	Create(int64, VariantCreateUpdate) (*VariantCreateUpdate, error)
+	Update(VariantCreateUpdate) (*VariantCreateUpdate, error)
```

Comment:
Can we just remove those two inventory properties from variant struct.
Two many duplicates between VariantCreateUpdate and Variant.

Reply:
That was my first approach but I remember having a discussion when I tried to do the same in bold-shopify-toolkit. Other projects were reading those variables. Shopify is still sending them in the GET but disallowing in POST and PUT. If we are sure no one is reading those variables I can go ahead and change the main model.

Reply:
Okay, that makes sense. We need keep it if the response need it.
However I don't think we need change anything.
There are omitempty tags in the struct, which means some of the properties should be omitted if it's not allowed. If you look at the update order request struct, other than `order_status`, `notes`, almost everything else can not be changed or posted in the update request.
I think the whole design doesn't separate requestObject and responseObject, which we might still want to follow.
If they do pass a inventory_quantity, we just need return the correct shopify error to tell them remove it.

Reply:
I do not think we need a new structure here. 
I believe we can keep the existing struct and zero-out deprecated fields, so they will be ignored because of `omitempty` attribute. This won't break existing code. 
Also we support API version parameters, so we can use it to have different behaviour for newer version of Shopify API
In Shopify deprecation notes they also noticed about `locationID` being required. Is it something we need to change as well?

Reply:
That seems to be a generic message from them, I already got rid of their warning in CSP and it was only by removing the inventory calls. We weren't making fulfillment or refund requests.

Reply:
I'll be closing this one guys, apparently not needed. It will be enough by anyone using this ignoring the variables when calling post/put